### PR TITLE
Rework pipeline component

### PR DESCRIPTION
**Pull request by timbussmann**
aligns the pipeline component better with the component design guidelines.

### CODE REVIEW

File: src/NServiceBus.Core/Pipeline/PipelineComponent.cs [hunk 1]
```diff
@@ -1,22 +1,37 @@
...
     using Extensibility;
     using ObjectBuilder;
     using Pipeline;
-    using Settings;
-    using System.Threading.Tasks;
 
     class PipelineComponent
     {
-        public PipelineComponent(SettingsHolder settings)
+        public PipelineComponent(PipelineModifications modifications)
```
Comment (andreasohlund) [inline src/NServiceBus.Core/Pipeline/PipelineComponent.cs]: Make this private?

**Comment by timbussmann** [comment]
made ctor private + rebased

File: src/NServiceBus.Core/EndpointConfiguration.cs [hunk 2]
```diff
@@ -53,7 +53,7 @@ public EndpointConfiguration(string endpointName)
         /// <summary>
         /// Access to the pipeline configuration.
         /// </summary>
-        public PipelineSettings Pipeline => PipelineComponent.PipelineSettings;
+        public PipelineSettings Pipeline { get; }
```
Comment (SeanFeldman) [inline src/NServiceBus.Core/EndpointConfiguration.cs]: I'd rather see `public PipelineSettings PipelineSettings { get; }` as the property represents pipeline settings, not the pipeline.

Comment (andreasohlund) [reply src/NServiceBus.Core/EndpointConfiguration.cs]: We can't do breaking changes at this stage. I think we would go with an extension method to align with all other APIs so:

`endpointConfiguration.Pipeline(p=>p.Register(...))`

Thoughts on that API?

Comment (SeanFeldman) [reply src/NServiceBus.Core/EndpointConfiguration.cs]: Sorry, I didn't realize that. The suggested API looks interesting. You'd have to show me more :)

**Pull request merged**